packages/react-menu/src/components/MenuItemCheckbox/useMenuItemCheckbox.ts

```diff
@@ -28,17 +29,8 @@ export const useMenuItemCheckbox = (
     resolveShorthandProps(props, menuItemCheckboxShorthandProps),
   );
 
-  useMenuItemSelectable(state, () => {
-    const newCheckedItems = [...state.checkedItems];
-    const index = state.checkedItems.indexOf(state.value);
-    if (index !== -1) {
-      newCheckedItems.splice(index, 1);
-    } else {
-      newCheckedItems.push(state.value);
-    }
-
-    return newCheckedItems;
-  });
+  const toggleCheckbox = useMenuListContext(context => context.toggleCheckbox);
+  useMenuItemSelectable(state, toggleCheckbox);
```

Comment:
```suggestion
  useMenuItemSelectable(state, 'checkbox');
```

What if `useMenuItemSelectable()` will accept a `kind` instead of a function directly?

Reply:
I'm not against it, but right now `useMenuItemSelectable` handles all the shared things for all selectables... accepting a `kind` breaks the encapsulation. 

Not saying that it would happen often (ever?) but in the future adding a new selectable item for `Menu` would result in a weird path where you need to know about the internals of this hook

Reply:
Agree, let's keep it as it is for now and check once more when an initial implementation will be done.